quil-rs/src/program/graph.rs

```diff
@@ -327,56 +327,54 @@ impl InstructionBlock {
                     Ok(())
                 }
                 InstructionRole::RFControl => {
-                    let used_frames = program
-                        .get_frames_for_instruction(instruction, false)
-                        .unwrap_or_default();
-
-                    let blocked_frames = program
-                        .get_frames_for_instruction(instruction, true)
-                        .unwrap_or_default();
-
-                    let blocked_but_not_used_frames = blocked_frames.difference(&used_frames);
-
-                    for frame in &used_frames {
-                        if instruction.is_scheduled() {
-                            let previous_node_ids = last_timed_instruction_by_frame
-                                .entry((*frame).clone())
-                                .or_default()
-                                .get_dependencies_for_next_user(node);
-
-                            for previous_node_id in previous_node_ids {
-                                add_dependency!(graph, previous_node_id => node, ExecutionDependency::Scheduled);
+                    let matched_frames = program.get_frames_for_instruction(instruction);
+
+                    if let Some(matched_frames) = matched_frames {
+                        if let Some(used_frames) = &matched_frames.used {
+                            for frame in used_frames {
+                                if instruction.is_scheduled() {
+                                    let previous_node_ids = last_timed_instruction_by_frame
+                                        .entry((*frame).clone())
+                                        .or_default()
+                                        .get_dependencies_for_next_user(node);
+
+                                    for previous_node_id in previous_node_ids {
+                                        add_dependency!(graph, previous_node_id => node, ExecutionDependency::Scheduled);
+                                    }
+                                }
+
+                                let previous_node_ids = last_instruction_by_frame
+                                    .entry((*frame).clone())
+                                    .or_default()
+                                    .get_dependencies_for_next_user(node);
+
+                                for previous_node_id in previous_node_ids {
+                                    add_dependency!(graph, previous_node_id => node, ExecutionDependency::StableOrdering);
+                                }
                             }
```

Comment:
I think it makes as much sense, if not more, to avoid the use of `Option` and use empty `HashSet`s for the cases where there are no matching frames.

Reply:
Performance benchmarks seem to indicate a 5% regression from this change alone, but I'll bump the sample size up and see that I can reproduce this.

Reply:
nvm - cannot reproduce now, it must have been due to something environmental. That's good news, since this is an ergonomic improvement.